### DISCUSSION TITLE
ESM repository not working

### DISCUSSION THREAD

**Discussion by Keysox** [renovatebot/renovate#18866] [Request Help]
When setting `"type": "module"` in `package.json` to enable ECMAScript Modules, Renovate no longer works.

In order to get access to logs, we typically use `--log-file` but in this instance, that file isn't being created.

Instead, we're receive the following error in stdout: `FATAL: Authentication failure`. Our CI system doesn't have access to the open internet. Any ideas how to fix or further debug?

**Comment by viceice** [comment]
do you try to run renovate as esm or do you try to update an esm repo with renovate?

**Reply by Keysox** [reply]
We've been running against this repo for months successfully but recently transitioned to using esm and it started failing. Do we have to run Renovate in a different way for it to work? I was hoping it would just read from that field and know what to do since it's still using npm to resolve dependencies

**Reply by viceice** [reply]
you should provide some more details and probably some debug logs parts from where the error occurs.

**Reply by viceice** [reply]
you seem to self-host renovate, so you probably got our latest major versions which had breaking changes.

**Reply by Keysox** [reply]
Correct, we're self-hosted and using latest. Was there a specific breaking change related to this? It's still working properly against numerous repositories not using esm

**Reply by viceice** [reply]
maybe, you should follow up our major releases an don't blindly run latest.

anyways without more details like logs or a reproduction we can't help. esm repos should work.

**Reply by Keysox** [reply]
And for some reason when it fails the log file isn't being persisted. Is there a way to run with debug logging enabled for standard out instead?

**Reply by viceice** [reply]
depends on how you run renovate, checkout our trouble shooting guide

https://docs.renovatebot.com/troubleshooting/

**Reply by Keysox** [reply]
Awesome, thank you for the link! I'm getting logs now and can investigate.

I didn't see `LOG_LEVEL` listed in https://docs.renovatebot.com/self-hosted-configuration/ so wasn't sure how to use that instead of the log file like we typically use, which isn't persisting in this scenario.

**Reply by Keysox** [reply]
When setting a repository to `type: module`, these are the relevant logs:
```
DEBUG: Parsing configs
DEBUG: Checking for config file in repo_org/repo_name/config.js
DEBUG: No config file found on disk - skipping
DEBUG: File config
       "config": {}
DEBUG: CLI config
       "config": {"repositories": ["repo_org/repo_name"]}
DEBUG: Env config
       "config": {"hostRules": [], "token": "***********"}
DEBUG: Combined config
       "config": {
         "hostRules": [],
         "token": "***********",
         "repositories": ["repo_org/repo_name"]
       }
DEBUG: Found valid git version
DEBUG: Using default github endpoint: https://api.github.com/
```

When I remove that line to not use esm, these are the relevant logs:
```
DEBUG: Parsing configs
DEBUG: Checking for config file in repo_org/repo_name/config.js
DEBUG: File config
       "config": {
         "endpoint": "https://private_github.com/api/v3/",
         "hostRules": [
           {
             "matchHost": "private_npm_registry.com",
             "username": "first.last",
             "password": "***********"
           },
         ],
         "password": "",
         "platform": "github",
         "username": ""
       }
DEBUG: CLI config
       "config": {"repositories": ["repo_org/repo_name"]}
DEBUG: Env config
       "config": {"hostRules": [], "token": "***********"}
DEBUG: Combined config
       "config": {
         "endpoint": "https://private_github.com/api/v3/",
         "hostRules": [
           {
             "matchHost": "private_npm_registry.com",
             "username": "first.last",
             "password": "***********"
           },
         ],
         "password": "",
         "platform": "github",
         "username": "",
         "token": "***********",
         "repositories": ["repo_org/repo_name"]
       }
DEBUG: Adding trailing slash to endpoint
DEBUG: Found valid git version
DEBUG: Detected GitHub Enterprise Server
```

Looking at https://github.com/renovatebot/renovate/blob/5e13228d52f74f0d2c25c9de61b254debecb694a/lib/workers/global/config/parse/file.ts#L11 I tried setting `RENOVATE_CONFIG_FILE=renovate.json`, these are the relevant logs:
```
DEBUG: Parsing configs
DEBUG: Checking for config file in repo_org/repo_name/renovate.json
DEBUG: File config
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": ["private_repo_org/private_repo_name"]
       }
DEBUG: CLI config
       "config": {"repositories": ["repo_org/repo_name"]}
DEBUG: Env config
       "config": {"hostRules": [], "token": "***********"}
DEBUG: Combined config
       "config": {
         "$schema": "https://docs.renovatebot.com/renovate-schema.json",
         "extends": ["private_repo_org/private_repo_name"],
         "hostRules": [],
         "token": "***********",
         "repositories": ["repo_org/repo_name"]
       }
DEBUG: Found valid git version
DEBUG: Using default github endpoint: https://api.github.com/
```

It seems like it isn't resolving the config properly once this field gets set. Any ideas? Thanks in advance!

**Reply by Keysox** [reply]
@viceice @rarkins any thoughts on the above? Appreciate the help, as always

**Reply by viceice** [reply]
it's not yet supported to use renovate inside esm projects

**Reply by Keysox** [reply]
Oh awesome, thanks! That helps. Would it be possible to have an issue tracking support for it if there's not already one out there I can follow?

**Reply by viceice** [reply]
search issues for esm, there is an open issue

**Reply by HonkingGoose** [reply]
> search issues for esm, there is an open issue

Here are the ESM issues for Renovate:

- #9890
- #13181
